### ISSUE TITLE
`Style/BitwisePredicate` autocorrect produce broken code

### ISSUE THREAD

**Issue by VitaliySerov** [rubocop/rubocop#13408]
## Expected behavior

Autocorrect shouldnt broke the code

## Actual behavior

```
$ ruby 1.rb 
1.rb:6:in `<main>': undefined method `allbits?' for an instance of Array (NoMethodError)

result = foo.allbits?(bar)
```

## Steps to reproduce the problem

1. Create a ruby script 1.rb

```ruby
foo = [1, 2, 3]
bar = [1, 2]

result = ((foo & bar) == bar)
print(result)
```

2. Execute it and make sure it returns `true`
3. `rubocop -A 1.rb` will produce:
```
# frozen_string_literal: true

foo = [1, 2, 3]
bar = [1, 2]

result = foo.allbits?(bar)
print(result)
```
3. run it again:
```ruby
$ ruby 1.rb 
1.rb:6:in `<main>': undefined method `allbits?' for an instance of Array (NoMethodError)

result = foo.allbits?(bar)
            ^^^^^^^^^

```

## RuboCop version

```
$ bundle exec rubocop -V
1.68.0 (using Parser 3.3.5.1, rubocop-ast 1.33.0, analyzing as Ruby 3.0, running on ruby 3.3.5) [x86_64-linux]
  - rubocop-performance 1.22.1
  - rubocop-rake 0.6.0
  - rubocop-rspec 3.2.0
```

**Comment by VitaliySerov** [comment]
@Earlopain  Fair enought, in my codebase using `intersection` method fix the problem (cop not triggered, and code not broken)

Not sure what to do with this issue though, close it or keep it open

**Comment by koic** [comment]
Since type-based analysis of the receiver isn’t possible, it is already marked as unsafe, and there is no precise way to resolve this false positive. This issue will be closed for now, to be reconsidered if a similar report is raised in the future. Thank you for the feedback.

**Comment by bbatsov** [comment]
Potentially we can improve the docs a bit to highlight better potential safety issues, but yeah - there's not that much we can do here.
